app/javascript/components/patient/FollowUpFlag.js

```diff
@@ -0,0 +1,283 @@
+      no_members_selected: false,
+      bulk_action_apply_to_household: false,
+      cancelToken: axios.CancelToken.source(),
+      clear_flag_disabled: true,
+      clear_flag: false,
+      clear_flag_reason: '',
+      follow_up_reason: '',
+      follow_up_note: '',
+      initial_follow_up_reason: '',
+      initial_follow_up_note: '',
+      loading: false,
+    };
+  }
+
+  componentDidMount() {
+    var state_updates = {};
+    if (this.props.bulk_action) {
+      // When the selected monitorees share the same follow up reason/note, pre-populate the modal with those values
+      const distinctFollowUpReason = [...new Set(this.props.patients.map(x => x.flagged_for_follow_up.follow_up_reason))];
+      const distinctFollowUpNote = [...new Set(this.props.patients.map(x => x.flagged_for_follow_up.follow_up_note))];
+
+      // If at least one monitoree has a follow up flag set, enable the clear flag option
+      if (!(distinctFollowUpReason.length === 1 && distinctFollowUpReason[0] === null)) {
+        state_updates.clear_flag_disabled = false;
+      }
+      if (distinctFollowUpReason.length === 1 && distinctFollowUpReason[0] !== null) {
+        state_updates.follow_up_reason = distinctFollowUpReason[0];
+      }
+      if (distinctFollowUpNote.length === 1 && distinctFollowUpNote[0] !== null) {
+        state_updates.follow_up_note = distinctFollowUpNote[0];
+      }
+
+      if (Object.keys(state_updates).length) {
+        this.setState(state_updates);
+      }
+    } else {
+      if (this.props.patient.follow_up_reason) {
+        // If the monitoree has a follow up flag set, enable the clear flag option
+        state_updates.clear_flag_disabled = false;
+        state_updates.follow_up_reason = this.props.patient.follow_up_reason;
+        state_updates.follow_up_note = this.props.patient.follow_up_note;
+        this.setState(state_updates);
+      }
+    }
+  }
+
+  handleChange = event => {
+    let value = event.target.type === 'checkbox' ? event.target.checked : event.target.value;
+    if (event.target.id === 'follow_up_reason') {
+      this.setState({ follow_up_reason: value });
+    } else if (event.target.id === 'follow_up_note') {
+      this.setState({ follow_up_note: value });
+    } else if (event.target.id === 'apply_to_household') {
+      this.setState({ apply_to_household: value });
+    } else if (event.target.id === 'bulk_action_apply_to_household') {
+      this.setState({ bulk_action_apply_to_household: value });
+    } else if (event.target.id == 'set_flag_for_follow_up') {
+      this.setState({ clear_flag: false });
+    } else if (event.target.id == 'clear_flag_for_follow_up') {
+      this.setState({ clear_flag: true });
+    } else if (event.target.id == 'clear_flag_reason') {
+      this.setState({ clear_flag_reason: value });
+    }
+  };
+
+  handleApplyHouseholdChange = apply_to_household => {
+    const no_members_selected = apply_to_household && this.state.apply_to_household_ids.length === 0;
```

Comment:
not sure this variable is necessary, you can just check `apply_to_household_ids.length === 0` where needed

Reply:
ah I see.  I'd request changing it to camel case however because this is native to the react component.  the only time we really use snake case on the front end is for props coming from an erb file or any variable going to the backend that has that same name in the schema

Reply:
Missed this in my most recent commit, I will include this tweak in the next one.